qlib/TableMapper.qm

```diff
@@ -597,32 +603,58 @@ table_mapper.setRowCode(rowcode);
+
+    list flushed_data = table_mapper.flush() ?? ();
+
+    mapped_data += flushed_data;
+    # mapped_data = ("num" : (1,2), "str" : ("bar","foo"));
+    # the table is updated too
 }
             @endcode
 
             Data is only inserted if the block buffer size reaches the limit defined by the \c "insert_block" option, in which case this method returns all the data inserted.  In case the mapped data is only inserted into the cache, no value is returned.
 
             @param rec the input record or record set in case a hash of lists is passed
-            @param crec an optional simple hash of data to be added to each row
+            @param crec an optional simple hash of data to be added to each input row before mapping
 
-            @return if batch data was inserted then a hash (columns) of lists (row data) of all data inserted and potentially returned (in case of sequences) from the database server is returned; if constant mappings are used with batch data, then they are returned as single values assigned to the hash keys
+            @return if batch data was inserted then a hash (columns) of lists (row data) of all data inserted and potentially returned (in case of sequences) from the database server is returned; if constant mappings are used with batch data, then they are returned as single values assigned to the hash keys; always returns a bulk (hash of lists)
 
             @note
             - make sure to call flush() before committing the transaction or discard() before rolling back the transaction or destroying the object when using this method
             - flush() or discard() needs to be executed for each mapper used in the block when using multiple mappers whereas the DB transaction needs to be committed or rolled back once per datasource
             - this method and batched inserts in general cannot be used when the \c "unstable_input" option is given in the constructor
-            - if the \c "insert_block" option is set to 1, then this method simply calls insertRow()
+            - if the \c "insert_block" option is set to 1, then this method simply calls insertRow(); even in this case a bulk is returned, though
             - if an error occurs flushing data, the count is reset by calling @ref Mapper::Mapper::resetCount() "Mapper::resetCount()"
-            - in case a hash of lists is passed in \a rec, then the the underlying @ref Mapper::Mapper "Mapper" object does not support structured input or output or XML CDATA in the input (input data in this format is assumed to come from an SQL query); note that this provides very high performance with SQL drivers that support Bulk DML
+            - in case a hash of lists is passed in \a rec, then the underlying @ref Mapper::Mapper "Mapper" object does not support structured input or output or XML CDATA in the input (input data in this format is assumed to come from an SQL query); note that this provides very high performance with SQL drivers that support Bulk DML
```

Comment:
this comment is not correct; batch or bulk mode processing does support structured input and output at least - I do not know about the cdata status